examples/ensemble/plot_hgbt_regression.py

```diff
@@ -0,0 +1,407 @@
+#   :ref:`sphx_glr_auto_examples_applications_plot_cyclical_feature_engineering.py`).
+#
+# Monotonic constraints
+# =====================
+#
+# Given specific domain knowledge that requires the relationship between a
+# feature and the target to be monotonically increasing or decreasing, one can
+# enforce such behaviour in the predictions of a HGBT model using monotonic
+# constraints. This makes the model more interpretable and can reduce its
+# variance (and potentially mitigate overfitting) at the risk of increasing
+# bias. Monotonic constraints can also be used to enforce specific regulatory
+# requirements, ensure compliance and align with ethical considerations.
+#
+# In the present example, the policy of transfering energy from Victoria to New
+# South Wales is meant to alleviate price fluctuations, meaning that the model
+# predictions have to enforce such goal, i.e. transfer should increase with
+# price and demand in New South Wales, but also decrease with price and demand
+# in Victoria, in order to benefit both populations.
+#
+# If the training data has feature names, it’s possible to specify the monotonic
+# constraints by passing a dictionary with the convention:
+#
+# - 1: monotonic increase
+# - 0: no constraint
+# - -1: monotonic decrease
+#
+# Else, one can pass an array-like encoding the above convention by position.
+
+from sklearn.inspection import PartialDependenceDisplay
+
+monotonic_cst = {
+    "date": 0,
+    "day": 0,
+    "period": 0,
+    "nswdemand": 1,
+    "nswprice": 1,
+    "vicdemand": -1,
+    "vicprice": -1,
+}
+hgbt_no_cst = HistGradientBoostingRegressor().fit(X, y)
+hgbt_cst = HistGradientBoostingRegressor(monotonic_cst=monotonic_cst).fit(X, y)
+
+fig, ax = plt.subplots(nrows=2, figsize=(15, 10))
+disp = PartialDependenceDisplay.from_estimator(
+    hgbt_no_cst,
+    X,
+    features=["nswdemand", "nswprice"],
+    line_kw={"linewidth": 2, "label": "unconstrained", "color": "tab:blue"},
+    ax=ax[0],
+)
+PartialDependenceDisplay.from_estimator(
+    hgbt_cst,
+    X,
+    features=["nswdemand", "nswprice"],
+    line_kw={"linewidth": 2, "label": "constrained", "color": "tab:orange"},
+    ax=disp.axes_,
+)
+disp = PartialDependenceDisplay.from_estimator(
+    hgbt_no_cst,
+    X,
+    features=["vicdemand", "vicprice"],
+    line_kw={"linewidth": 2, "label": "unconstrained", "color": "tab:blue"},
+    ax=ax[1],
+)
+PartialDependenceDisplay.from_estimator(
+    hgbt_cst,
+    X,
+    features=["vicdemand", "vicprice"],
+    line_kw={"linewidth": 2, "label": "constrained", "color": "tab:orange"},
+    ax=disp.axes_,
+)
+_ = plt.legend()
+
+# %%
+# Indeed, we can verify that the predictive quality of the model is not
+# significantly degraded by introducing the monotonic constraints:
+
+from sklearn.model_selection import TimeSeriesSplit, cross_validate
+
+ts_cv = TimeSeriesSplit(n_splits=5, gap=48, test_size=336)
+
+cv_results = cross_validate(
```

Comment:
This is all very heavy (and correct). Simply using the test set that we already have could be enough.

Actionable: Also print the simple test set numbers and spend one sentence as to why using `cross_validate`.

Reply:
Addressed in [9499e61](https://github.com/scikit-learn/scikit-learn/pull/26991/commits/9499e611a7b2159c0685a0195cdaad37902f55ac).